content/blog/2018/02/2018-02-22-cheetah.adoc

```diff
@@ -0,0 +1,134 @@
+
+image:/images/post-images/2018-02-22-cheetah/pipeline-durability-highlighted.png[role="center"]
+
+[pipeline]
+----
+// Script //
+properties([durabilityHint('PERFORMANCE_OPTIMIZED')])
+// Declarative //
+pipeline {
+    agent any
+    stages {
+        stage('Example') {
+            steps {
+                echo 'Hello World'
+            }
+        }
+    }
+    options {
+        durabilityHint('PERFORMANCE_OPTIMIZED')
+    }
+}
+----
+
+=== 3. **Multibranch Projects** can use a new BranchProperty to customize the Durability Setting.
+Under the SCM you can configure a custom Branch Property Strategy and add a property for Custom Pipeline Speed/Durability Level.  This overrides the global Durability Setting and will apply to each branch at the next run.  You can also use a "properties" step to override the setting, but remember that you may have to run the step again to undo this.
+
+image:/images/post-images/2018-02-22-cheetah/multibranch-durability.png[role="center"]
+
+Durability settings will take effect with the next applicable Pipeline run, not immediately.  The setting will be displayed in the log. 
+
+image:/images/post-images/2018-02-22-cheetah/durability-in-log-highlighted.png[role="center"]
+
+
+== Will Performance-Optimized Mode Help Me?
+* Yes, if your Jenkins instance uses NFS, magnetic storage, runs many Pipelines at once, or shows high iowait (above 5%)
+* Yes, if you are running Pipelines with many steps (more than several hundred).
+* Yes, if your Pipeline stores large files or complex data to variables in the script, keeps that variable in scope for future use, and then runs steps.  This sounds oddly specific but happens more than you'd expect.
+** For example: `readFile` step with a large XML/JSON file, or using configuration information from parsing such a file with link:https://jenkins.io/doc/pipeline/steps/pipeline-utility-steps/#code-readjson-code-read-json-from-files-in-the-workspace[One of the Utility Steps].
+** Another common pattern is a "summary" object containing data from many branches (logs, results, or statistics). Often this is visible because you'll be adding to it often via an add/append or `Map.put()` operations.
+** Large arrays of data or Maps of configuration information are another common example of this situation.
+* No, if your Pipelines spend almost all their time waiting for a few shell/batch steps to finish.  This ISN'T a magic "go fast" button for everything!
+* No, if Pipelines are writing massive amounts of data to logs (logging is unchanged).
+* No, if you are not using Pipelines, or your system is loaded down by other factors.
+* No, if you don't enable higher-performance modes for pipelines.
+
+== Other Goodies
+
+* Users can now set a job property so that individual Pipelines fail cleanly rather than resuming upon restarting the master. 
```

Comment:
Having spent a _lot_ of my time advocating for Pipeline and "durablility" as part of it, I feel like this needs a little more "meat on the bones" worth of explanation. What are the benefits of failing cleanly rather than resuming?

Reply:
Good point, clarifying a bit -- this is a somewhat less common use case (though it has been requested a couple times for various reasons)